salesagility/SuiteCRM
# Improve Vardef Length Comparison

## Description
Allows increasing of Length/Precision in accordance with the data safe logic below, and works on Int/Float/Decimal 

## Motivation and Context
The previous solution ignored changes to the scale (digits to the right of the decimal point)

## How To Test This
Make a change in either studio or vardefs to a Decimal or Float fields.

Changes to the "len" value will only be detected when running a quick repair and rebuild if the first value is changed
eg; OOTB, 'tax_amount' for 'AOS_Quotes' has a default 'len' value of '26,6'

--Changing to 27,6 is detected, get SQL on R&R
--Changing to 27,7 is detected, get SQL on R&R
--Changing to 26,7 is not detected, no SQL on R&R
--Changing to 25,6 is not detected, no SQL on R&R

To add some context, it is not possible to go from 26,6 to 26,7 safely, without checking the existing data set for length so this change is ignored for now. I believe the database performs this check on the data when the SQL is executed but the aim was to prevent the R&R process providing SQL which may return an error.

## Types of changes

### Final checklist
